src/core/autotests/testmasterkey.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+   SPDX-FileCopyrightText: 2025 Andro Ranogajec <ranogaet@gmail.com>
+
+   SPDX-License-Identifier: LGPL-2.0-or-later
+*/
+
+#include "testmasterkey.h"
+#include "encryption/encryptionutils.h"
+#include <QDebug>
+#include <QTest>
+
+TestMasterKey::TestMasterKey(QObject *parent)
+    : QObject(parent)
+{
+}
+
+/**
+ * @brief This methods to going to test the determinism of the master key.
+ *
+ * if
+ * n == n1 and m == m1
+ * then
+ * getMasterKey(n, m) == getMasterKey(n1, m1)
+ *
+ * if
+ * n != n1 and m != m1
+ * then
+ * getMasterKey(n, m) != getMasterKey(n1, m1)
+ */
+void TestMasterKey::testMasterKeyDeterminism()
+{
+    QString masterKey1;
+    QString masterKey2;
+
+    for (int i = 0; i <= 10; i++) {
+        masterKey1 = EncryptionUtils::getMasterKey(QStringLiteral("admin"), QStringLiteral("root"));
+        masterKey2 = EncryptionUtils::getMasterKey(QStringLiteral("admin"), QStringLiteral("root"));
+        QVERIFY(masterKey1 == masterKey2);
+
+        masterKey1 = EncryptionUtils::getMasterKey(QStringLiteral("123"), QStringLiteral("abc"));
+        masterKey2 = EncryptionUtils::getMasterKey(QStringLiteral("abc"), QStringLiteral("321"));
+        QVERIFY(masterKey1 != masterKey2);
+    }
+}
+
+void TestMasterKey::testMasterKeyEmptyPassword()
+{
+    QString masterKey = EncryptionUtils::getMasterKey(QStringLiteral(""), QStringLiteral("someUser"));
+    QVERIFY(masterKey.isEmpty());
+}
+
+void TestMasterKey::testMasterKeyEmptyUserId()
+{
+    QString masterKey = EncryptionUtils::getMasterKey(QStringLiteral("somePassword"), QStringLiteral(""));
+    QVERIFY(masterKey.isEmpty());
+}
+
+void TestMasterKey::testImportRawKey()
+{
+}
+
+QTEST_GUILESS_MAIN(TestMasterKey)
```

Comment:
move it at line 11
=> when we add more test function this line is not moved all the time (better for diff)